controllers/hivelocitycluster_controller.go

```diff
@@ -477,6 +477,10 @@ func reconcileTargetSecret(ctx context.Context, clusterScope *scope.ClusterScope
 }
 
 func (r *HivelocityClusterReconciler) reconcileTargetClusterManager(ctx context.Context, clusterScope *scope.ClusterScope) (res reconcile.Result, err error) {
+	if !conditions.IsTrue(clusterScope.Cluster, clusterv1.ControlPlaneInitializedCondition) {
```

Comment:
I don't think this is needed. Please check out the PR of @aniruddha2000 and CAPH if something is not aligned yet. We should use the same approach as we have there.

Reply:
I removed that code.